### PR TITLE
update get diffs API to support summary mode

### PR DESCRIPTION
**Pull request by mrdadah**

### CODE REVIEW

File: src/endpoints/get-diffs.js [hunk 1]
```diff
@@ -69,6 +55,6 @@ module.exports = cors(async (req, res) => {
     patch,
     changeLog,
     latestVersion: session.summary_version,
-    hashOfLatestState: hash(session.udt_json),
```
Comment (seveibar) [inline src/endpoints/get-diffs.js:72]: this hash probably needs to be merged with another hash, something like `hash(session.summary_samples) + hash(session.interface)`

Comment (seveibar) [reply src/endpoints/get-diffs.js:72]: otherwise interface changes wouldn't be captured (unless, is interface included in summary_samples?)

Comment (mrdadah) [reply src/endpoints/get-diffs.js:72]: @seveibar yes interface included in summary_samples?
maybe the naming is confusing, should I change it?

Comment (mrdadah) [reply src/endpoints/get-diffs.js:72]: summary_samples will contain this object
`{
  "interface": { /* ... */ },
  "summary": {
    "samples": [
      { "state": "complete", version: 32 }
      // ...
    ]
  }
}`

Comment (seveibar) [reply src/endpoints/get-diffs.js:72]: gotcha, yea consider a rename in a future PR. "summary_object" is probably better

**Pull request merged**